### ISSUE TITLE
Review ComboBox 'autoHide' design

### ISSUE THREAD

**Issue by fergusonr** [gui-cs/Terminal.Gui#820]
Background: ComboBox was initially written with one use-case in mind (hosted in a dialog), @BDisp then helped extend it to a more generalized control. Of which there are now two render modes. 1. List hidden until user types or hits drop down arrow, and 2. List is always displayed. I aimed to auto-detect which mode should be used which led to the introduction of the Added() Removed() actions.

I believe the simple answer to this is to not try and auto-detect which mode to use and make the mode propety 'autoHide' public so the user can decided. And in doing so, change the name and invert the logic
<pre>
from:
    autoHide = true

to:
    AlwaysShow = false
</pre>

In summary, the complexity is not down to this being a composite control, it's down to my own making :)

**Comment by BDisp** [comment]
I added a `ReadOnly` property to the `ComboBox` for not allowing user typing which is working as expected.. See https://github.com/migueldeicaza/gui.cs/pull/1385#issuecomment-891280970

Maybe later if the `autoHide` will have unexpected behaviors, then will need to create a public `AutoHide` property to force the user choice.
